PluginBuilder/Controllers/ApiController.cs

```diff
@@ -60,16 +60,50 @@ public async Task<IActionResult> Plugins(
         };
         await using var conn = await connectionFactory.Open();
 
+        var filters = new List<string>
+        {
+            "b.manifest_info IS NOT NULL",
+            "b.build_info IS NOT NULL"
+        };
+
+        var isBtcpayV21OrHigher = btcpayVersion?.IsAtLeast(2, 1) == true;
+        var hasSearchTerm = !string.IsNullOrWhiteSpace(searchPluginName);
+
+        if (isBtcpayV21OrHigher)
+        {
+            if (hasSearchTerm)
+            {
+                filters.Add("(p.visibility = 'listed' OR p.visibility = 'unlisted')");
+                filters.Add("(p.slug ILIKE @searchPattern OR b.manifest_info->>'Name' ILIKE @searchPattern)");
+            }
+            else
+            {
+                filters.Add("p.visibility = 'listed'");
+            }
+        }
+        else
+        {
+            filters.Add("(p.visibility = 'listed' OR p.visibility = 'unlisted')");
+            if (hasSearchTerm)
+            {
+                filters.Add("(p.slug ILIKE @searchPattern OR b.manifest_info->>'Name' ILIKE @searchPattern)");
+            }
+        }
+
+        if (!string.IsNullOrWhiteSpace(searchPluginIdentifier))
+        {
+            filters.Add("p.identifier = @searchPluginIdentifier");
+        }
```

Comment:
@thgO-O, one notice here - @NicolasDorier has changed the logic in BTCPay Server recently in order to have `searchPluginIdentifier` variable to fetch plugin information regardless of visibility for autoinstall. 

Check out: https://github.com/btcpayserver/btcpayserver/pull/6858/files

Before merging PR you and me should discuss how to best integrate this considering all requirements. It could be even a new endpoint - to keep this method backwards compatible and code much cleaner.